trie/iterator.go

```diff
@@ -31,19 +32,25 @@ type Iterator struct {
 // NewIterator creates a new key-value iterator.
 func NewIterator(trie *Trie) *Iterator {
 	return &Iterator{
-		trie:   trie,
 		nodeIt: NewNodeIterator(trie),
-		keyBuf: make([]byte, 0, 64),
+		Key:    nil,
```

Comment:
Ha, just saw this was in the original code too. Anyway, probably should be deleted.